plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -899,11 +925,273 @@ public void testForceRecalculateAllStats()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testNoStats()
+            throws Exception
+    {
+        String tableName = copyResourcesAndRegisterTable("no_stats");
+        assertQuery(
+                "SELECT * FROM " + tableName,
+                """
+                        VALUES
+                        (42, 'foo'),
+                        (12, 'ab'),
+                        (null, null),
+                        (15, 'cd'),
+                        (15, 'bar')
+                        """);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('c_int', null, null, null, null, null, null),
+                        ('c_str', null, null, null, null, null, null),
+                        (null, null, null, null, null, null, null)
+                        """);
+
+        assertUpdate("ANALYZE " + tableName, 5);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                """
+                        VALUES
+                        ('c_int', null, 3.0, 0.2, null, 12, 42),
+                        ('c_str', 10.0, 4.0, 0.2, null, null, null),
+                        (null, null, null, null, 5.0, null, null)
+                        """);
+
+        // Ensure that ANALYZE does not change data
+        assertQuery(
+                "SELECT * FROM " + tableName,
+                """
+                        VALUES
+                        (42, 'foo'),
+                        (12, 'ab'),
+                        (null, null),
+                        (15, 'cd'),
+                        (15, 'bar')
+                        """);
```

Comment:
- The expected can be one line (redundant multiline string)
- extract expected as a variable to emphasize state didn't change between start and end of the test